tests/collections/llm/gpt/data/test_chat_dataset.py

```diff
@@ -24,12 +24,102 @@
 from nemo.collections.llm.gpt.data.chat import ChatDataModule
 
 
+def mock_llama_apply_chat_template(
```

Comment:
can you add a test that checks that the final tokens after `apply_chat_template` are correct? otherwise I don't see the purpose of adding in a mock `apply_chat_template`.

 instead of a mock function we could also add a HF tokenizer for Llama 3.2 1B in the test files and load that

Reply:
Added another test that uses the llama 3 tokenizer, validates the tokens